templates/c/src/tic80.h

```diff
@@ -0,0 +1,262 @@
+//if color == -1 this should return color
+
+WASM_IMPORT("rect")
+void rect(int32_t x, int32_t y, int32_t w, int32_t h, uint8_t color);
+
+WASM_IMPORT("rectb")
+void rectb(int32_t x, int32_t y, int32_t w, int32_t h, uint8_t color);
+
+WASM_IMPORT("spr")
+void spr(int32_t id, int32_t x, int32_t y, uint8_t* trans_colors, int32_t colorCount,
+	 int32_t scale, int32_t flip, int32_t rotate, int32_t w, int32_t h);
+
+WASM_IMPORT("tri")
+void tri(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+	 int32_t x3, int32_t y3, uint8_t color);
+
+WASM_IMPORT("trib")
+void trib(int32_t x1, int32_t y1, int32_t x2, int32_t y2,
+	  int32_t x3, int32_t y3, uint8_t color);
+
+WASM_IMPORT("textri")
+void textri(int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t x3, int32_t y3,
+	    int32_t u1, int32_t v1, int32_t u2, int32_t v2, int32_t u3, int32_t v3,
+	    bool use_map, uint8_t* trans_colors, int8_t colorCount);
+
+
+// ----
+// TEXT
+
+// print text [x=0 y=0] [color=15] [fixed=false] [scale=1] [smallfont=false] -> text width
+WASM_IMPORT("print")
+int32_t print(const char *text, int32_t x, int32_t y,
+	      int32_t color, bool fixed, int32_t scale, int32_t smallfont);
+// font text, x, y, [transcolor], [char width], [char height], [fixed=false], [scale=1] -> text width
+WASM_IMPORT("font")
+int32_t font(const char *text, uint32_t x, int32_t y, int32_t transcolor,
+	 int32_t char_width, int32_t char_height, bool fixed, int32_t scale);
+
+// -----
+// AUDIO
+
+WASM_IMPORT("music")
+void music(int32_t track, int32_t frame, int32_t row,
+	   bool loop, bool sustain, int32_t tempo, int32_t speed);
+// sfx id [note] [duration=-1] [channel=0]
+//        [volumeleft=15] [volumeright=15] [speed=0]
+WASM_IMPORT("sfx")
+void sfx(int32_t id, int32_t note, int32_t duration, int32_t channel,
+	 int32_t volume_left, int32_t volume_right, int32_t speed);
+
+
+// ------
+// MEMORY
+
+// pmem index -1 -> val
+// pmem index val -> val
+WASM_IMPORT("pmem")
+uint32_t pmem(int32_t index, uint32_t value);
+WASM_IMPORT("memcpy")
+void tic_memcpy(uint32_t to, uint32_t from, uint32_t length);
+WASM_IMPORT("memset")
+void tic_memset(uint32_t addr, uint8_t value, uint32_t length);
+
+WASM_IMPORT("poke")
+void poke(int32_t addr, uint8_t value, int32_t bits);
+WASM_IMPORT("poke4")
+void poke4(int32_t addr4, uint8_t value);
+WASM_IMPORT("poke2")
+void poke2(int32_t addr2, uint8_t value);
+WASM_IMPORT("poke1")
+void poke1(int32_t bitaddr, uint8_t value);
+
+WASM_IMPORT("peek")
+uint8_t peek(int32_t addr, int32_t bits);
+WASM_IMPORT("peek4")
+uint8_t peek4(int32_t addr4);
+WASM_IMPORT("peek2")
+uint8_t peek2(int32_t addr2);
+WASM_IMPORT("peek1")
+uint8_t peek1(int32_t bitaddr);
+
+WASM_IMPORT("vbank")
+int32_t vbank(int32_t bank);
+
+
+// -----
+// UTILS
+
+// time -> ticks (ms) elapsed since game start
+WASM_IMPORT("time")
+float time();
+
+WASM_IMPORT("tstamp")
+uint64_t tstamp();
+
+WASM_IMPORT("sync")
+void sync(int32_t mask, int8_t bank, int8_t tocart);
+// TODO should tocart be bool?
```

Comment:
It's not to allow passing -1 for "default"... did you follow the whole API discussion on GitHub?  There was some question about how much work the WASM layer should do in so far as proving default values so someone could call a function like:

```js
// C
sync(-1,-1,-1) 
// Lua
sync()
```

Reply:
If we finally said "no defaults" sorry, then yes should be bool, but I think since this is a function that typically allows var args that we decided one should be able to use -1 as a placeholder for "i'm not providing this arg".

Reply:
OK I see, didn't realize it was optional.